source/common/stats/statsd.cc

```diff
@@ -97,12 +97,55 @@ TcpStatsdSink::TlsSink::~TlsSink() {
   }
 }
 
+void TcpStatsdSink::TlsSink::beginFlush(bool expect_empty_buffer) {
+  ASSERT(!expect_empty_buffer || buffer_.length() == 0);
+  ASSERT(current_slice_mem_ == nullptr);
+  UNREFERENCED_PARAMETER(expect_empty_buffer);
+
+  uint64_t num_iovecs = buffer_.reserve(FlushSliceSizeBytes, &current_buffer_slice_, 1);
+  ASSERT(num_iovecs == 1);
+  UNREFERENCED_PARAMETER(num_iovecs);
```

Comment:
I clearly didn't think this one through completely, since it's an expression not a variable. That's what I get for commenting quickly before going to get coffee rather than the other way around.

When NDEBUG, ASSERT _could_ assign the expression to a boolean that is itself UNREFERENCED_PARAMETER. If the expression cannot have side effects (like above) I think it should be optimized out but if the expression has side effects it won't really get compiled out like we want.

An attempt at a quick experiment reveals plenty of gumption traps so forget I ever said this :)